spec/parser_spec.rb

```diff
@@ -5,10 +5,45 @@
 describe Parser do
   it { should respond_to(:parse) }
 
-  let(:document) { File.read('spec/fixtures/valid_result.xml') }
-
   describe "#parse" do
     subject { Parser.new.parse(document) }
-    it { should be_a Result }
+
+    context "with a valid result" do
+      let(:document) { Fixture.get(:valid) }
+
+      it { is_expected.to be_a Result }
+    end
+
+    context "with a nonexistent api key result" do
+      let(:document) { Fixture.get(:nonexistent_api_key) }
+
+      it "should raise AuthError" do
+        expect { subject }.to raise_error(AuthError)
+      end
+    end
+
+    context "with a wrong api key result" do
```

Comment:
What we really mean with `wrong` is that they are using an API key for one of our other products, can we get that into the name (and fixture) without making it too wordy it would be great